### PR TITLE
Change image repository  (#1)

### PR DESCRIPTION
**Pull request by wenterjoy**
* change image repository by configmap

* change plugin framework to common

* add changelogs

* add comments to changlog
Signed-off-by: wenterjoy <wenterjoy@126.com>

Thank you for contributing to Velero!

# Please add a summary of your change

add new RestoreItemAction of "velero.io/change-image-repository" to handle the issue mentioned at #5519
"considering the local image repositories changes when migrating from cluster A to B"

# Does your change fix a particular issue?
#5519
Fixes #(issue)
#5519
# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Comment by blackpiglet** [comment]
@wenterjoy 
Velero v1.10 is under testing for RC.
May need to put new function PR on hole, before release branch cut.

File: changelogs/unreleased/5519-wenterjoy [hunk 1]
```diff
@@ -0,0 +1,34 @@
+add new RestoreItemAction of "velero.io/change-image-repository" to handle the issue mentioned at #5519
+
+
```
Comment (sseago) [inline changelogs/unreleased/5519-wenterjoy:3]: Documentation doesn't really belong in the changelog entries. Just keep the above line and everything below should be moved to docs. You'll want to add a section to site/content/docs/main/restore-reference.md similar to the existing "Changing PV/PVC Storage Classes" section.

File: pkg/restore/change_image_repository_action.go [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+		logger:          logger,
+		configMapClient: configMapClient,
+	}
+}
+
+// AppliesTo returns the resources that ChangeImageRepositoryAction should
+// be run for.
+func (a *ChangeImageRepositoryAction) AppliesTo() (velero.ResourceSelector, error) {
+	return velero.ResourceSelector{
+		IncludedResources: []string{"deployment", "statefulsets", "daemonset", "pod"},
```
Comment (sseago) [inline pkg/restore/change_image_repository_action.go:59]: For completeness, this should also include ReplicaSet, ReplicationController, Job, and CronJob

File: changelogs/unreleased/5519-wenterjoy (hunk 1 above)
Comment (wenterjoy) [reply changelogs/unreleased/5519-wenterjoy:3]: thanks for your comments and advice

**Comment by wenterjoy** [comment]
> @wenterjoy Velero v1.10 is under testing for RC. May need to put new function PR on hole, before release branch cut.

@blackpiglet  Ok, thanks for reminder. I will create new function PR ASAP. So should me close this one？

File: changelogs/unreleased/5519-wenterjoy (hunk 1 above)
Comment (wenterjoy) [reply changelogs/unreleased/5519-wenterjoy:3]: @sseago  site/content/docs/main/restore-reference.md updated.

**Pull request closed without merging**